src/main/java/com/udojava/evalex/Expression.java

```diff
@@ -1899,4 +1901,20 @@ public boolean isBoolean() {
 		return false;
 	}
 
+	public List<String> infixNotation() {
+		final List<String> infix = new ArrayList<String>();
+		Tokenizer tokenizer = new Tokenizer(expression);
+		while (tokenizer.hasNext()) {
+			Token token = tokenizer.next();
+			StringBuilder sb = new StringBuilder();
+			sb.append("{")
+				.append(token.type)
+				.append(":")
+				.append(token.surface)
+				.append("}");
+			infix.add(sb.toString());
```

Comment:
You don't require a `StringBuilder` for simple concatenations, as both the bytecode compiler and the JIT compiler will optimize calls like these:

~~~java
String a = b + c + d + e;
~~~

Which will already compile to:

~~~
String a = new StringBuilder().append(b).append(c).append(d).append(e);
~~~

You only *require* a `StringBuilder` when building Strings *over* a loop. For example:

~~~java
StringBuilder string = new StringBuilder();

for (Item item in lotsOfItems) {
    string.append(item.toString());
}

return string.toString();
~~~

Because if you do this with normal String operations, the memory of the whole String will be copied on every iteration.